source/common/router/router.cc

```diff
@@ -738,6 +778,47 @@ void Filter::sendNoHealthyUpstreamResponse() {
                              StreamInfo::ResponseCodeDetails::get().NoHealthyUpstream);
 }
 
+/*
+ * Helper class to handle a known number of buffer copies.
+ */
+class ManyCopiedBuffer {
```

Comment:
Can this be placed with the other buffer code?

Reply:
Done.